### PR TITLE
Update to .NET 6 Preview 6 and use crossgen2

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

- Update to .NET 6 Preview 6
- Update to use crossgen2 instead of crossgen as crossgen.exe is not available in preview 6.
- Fix CA1845 issues
- Add workaround for issues https://github.com/dotnet/sdk/issues/18995 and https://github.com/dotnet/sdk/issues/18999
- Disable Test-Connection MTUSize tests due to https://github.com/dotnet/runtime/issues/55961
- Disable tests due to https://github.com/dotnet/runtime/issues/54251
- Update startup test to remove `System.IO.FileSystem.dll` as it is not loaded anymore.

## PR Context

Update to use crossgen2 instead of crossgen

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: build.psm1 [hunk 1]
```diff
@@ -2323,78 +2323,79 @@ function Start-CrossGen {
...
+        # The path to folder must end with directory separator
+        $dirSep = [System.IO.Path]::DirectorySeparatorChar
+        $platformAssembliesPath = if (-not $platformAssembliesPath.EndsWith($dirSep)) { $platformAssembliesPath + $dirSep }
+
+        Start-NativeExecution {
+            $crossgen2Params = @(
+                "-r"
+                $platformAssembliesPath
+                "--out-near-input"
+                "--single-file-compilation"
```
Comment (iSazonov) [inline build.psm1:2388]: Do we switch to single file?

**Comment by adityapatwardhan** [comment]
Please fix CI issues.

File: build.psm1 (hunk 1 above)
Comment (adityapatwardhan) [reply build.psm1:2389]: This is for crossgen'ing the input files individually. Not related to PublishSingleFile

File: build.psm1 [hunk 2]
```diff
@@ -2403,24 +2412,11 @@ function Start-CrossGen {
...
-            $generateSymbols = $true
-        } elseif (!($env:PROCESSOR_ARCHITECTURE -match "arm")) {
-            throw "crossgen for 'win-arm' and 'win-arm64' must be run on that platform"
-        }
-    } elseif ($Runtime -eq "linux-arm") {
-        throw "crossgen is not available for 'linux-arm'"
-    } elseif ($Runtime -eq "linux-x64") {
-        $Runtime
-        # We should set $generateSymbols = $true, but the code needs to be adjusted for different extension on Linux
+        "win-x64"
```
Comment (TravisEz13) [inline build.psm1:2419]: Can you comment why?

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 3]
```diff
@@ -261,7 +261,8 @@ Describe "Test-Connection" -tags "CI" {
     Context "MTUSizeDetect" {
         # We skip the MtuSize detection tests when in containers, as the environments throw raw exceptions
         # instead of returning a PacketTooBig response cleanly.
-        It "MTUSizeDetect works" -Pending:($env:__INCONTAINER -eq 1) {
+        # Test disabled due to .NET runtime issue: https://github.com/dotnet/runtime/issues/55961
+        It "MTUSizeDetect works" -Pending:(($env:__INCONTAINER -eq 1) -or $true) {
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:265]: disable only for macOS

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments

**Review by TravisEz13 — APPROVED** [review]
Approved with comments to be addressed

File: build.psm1 (hunk 2 above)
Comment (adityapatwardhan) [reply build.psm1:2419]: Added comment

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 (hunk 3 above)
Comment (adityapatwardhan) [reply test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1:265]: Fixed

**Pull request merged**